### DISCUSSION TITLE
Drive Strength of Wire/I2C

### DISCUSSION THREAD

**Discussion by ZinnerC** [earlephilhower/arduino-pico#1930] [Q&A] [answered]
I have currently a circuit which shifts the I2C@3.3v to 5v. For this purpose I use the TCA9517.
This setup had the ocaional issue that the I2C got killed. This because of the electrical circuit used (see below if interested).
While checking I saw that, if I would set the I2C to a drive strength of 8mA it should actually all be good. The TCA9517 has a max of 6mA on the side of the Pico (low level, see datasheet in 7.3)

The question now is how I set the drive strength for the I2C. For a general output it is clear ([documentation](https://arduino-pico.readthedocs.io/en/latest/digital.html)) however for the I2C it is not.

Would be nice if somebody could give me a hint who I could set the drive strength.

TCA9517: https://www.ti.com/lit/ds/symlink/tca9517.pdf

For thos interested:
![TCA9517 on proxy](https://github.com/earlephilhower/arduino-pico/assets/7002769/8fd8008c-3628-40e0-8a73-ff1d106e78c7)

**Comment by earlephilhower** [comment]
I think just setting the `pinMode(<pin>, OUTPUT_8MA)` should work.  There's a mux in front of the pad, and that's what I2C overrides.  It should not override the pad control AIUI.

**Reply by ZinnerC** [reply]
Thanks. I will give it a try. There is not a way to check that by any means?

**Reply by earlephilhower** [reply, marked as answer]
Apart from physical testing (Trc, for example) you can read the RP2040 chip datasheet.  IT has the muxing and pad configurations.  I'm pretty sure the drive strength is independent of what is driving the pin.

You should also manually call the gpio_Set_drive_strength (see cores/rp2040/wiring_digital.cpp) after starting the SPI/Wire.begin just to be double-certain.
